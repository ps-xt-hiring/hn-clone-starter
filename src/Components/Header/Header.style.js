/* eslint-disable */
import styled from 'styled-components';
import Colors from '../../constants/theme';

const HeaderContainer = styled.div`
  background-color: ${Colors.headerBackground};
  display: flex;
  justify-content: flex-start;
  align-items: center;
`;

export default HeaderContainer;
