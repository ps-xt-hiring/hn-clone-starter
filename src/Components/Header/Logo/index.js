/* eslint-disable */
export { default } from './Logo';
