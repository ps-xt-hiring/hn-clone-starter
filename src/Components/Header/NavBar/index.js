/* eslint-disable */
export { default } from './NavBar';
