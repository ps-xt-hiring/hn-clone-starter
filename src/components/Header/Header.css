.news-header {
    background-color: #fa6602;
    height: 1vh;
    display: flex;
    align-items: center;
    padding: 1em;
  }
  
  .news-header .logo {
    border: 1px solid white;
    color: white;
    padding: 0.2em;
  }
  
  .news-header .topLink {
    color: white;
    padding: 0.6em;
  }
  
  .news-header .newLink {
    color: black;
    border-left: 1px solid white;
    padding-left: 0.6em;
  }
  
  .news-header a {
    text-decoration: none;
  }
  