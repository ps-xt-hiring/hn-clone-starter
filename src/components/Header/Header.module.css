.mainDiv {
    height: 24px;
    background-color: #ff6600;
}

.headerTable {
    border: "0";
    width: "100%";
    padding: 2px
}

.tableData {
    width: 18px;
    padding-right: 4px;
}
.frontPageImg{
    width:18px;
    height:18px;
    border:1px white solid;
    padding-right: 4px;
    

}
.lifrontPageImg{
    padding-left: 2px;
    padding-top: 2px;
}

ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
    overflow: hidden;
    padding-bottom: 4px;
  }
  
  li {
    float: left;
  }
  
  li a {
    display: block;
    color: #000000;
    text-align: center;
    padding: 2px 16px;
    text-decoration: none;
  }
  
  
  .liStyle{
    margin-left: -26px;
  }
  .liStylelast{
      float: right;
  }