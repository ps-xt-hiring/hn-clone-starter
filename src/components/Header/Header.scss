@import '../../helpers/_utils.scss';

.header {
    background-color : #ff6600;
    font-size: .9rem;
    width: 100%;
    margin: 0 auto;
    
    .logo{
      border: 1px solid white;
      width: 20px;
      height: 20px;
    }
  
    a {
      text-decoration: none;      
    }
    a:hover {
      text-decoration: none;
    }

  }
  

