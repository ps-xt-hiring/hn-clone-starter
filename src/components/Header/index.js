import React from 'react';
import { HeaderSection, Logo } from '../Styled';
import logoSrc from '../../Assets/y18.gif';

const Header = () => (
  <HeaderSection className="App-header">
    <Logo src={logoSrc} alt="Logo" />
    <nav>
      <button type="button" className="active">top</button>
      <button type="button" className="excludeBorder">new</button>
    </nav>
  </HeaderSection>
);

export default Header;
