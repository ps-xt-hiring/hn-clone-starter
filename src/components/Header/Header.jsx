import React from 'react';

import Logo from '../../assets/y18.gif';
import TEXT from '../../constants/text.constants';
import './Header.scss';

const Header = () => (
    <>
      <div className='header'>
        <span>
          <img className='logo m-lr-2' src={Logo} alt={TEXT.HACKER_NEWS} />
        </span>
        <a className='m-tb-2 whiteText' href="http://www.hackernews.com" title="Hacker News">
          {TEXT.TOP}
        </a>
          &nbsp;|&nbsp;
        <a className='m-tb-2 darkGreyText' href="http://www.hackernews.com" title="Hacker News" >
          {TEXT.NEW}
        </a>
      </div>
    </>
  );

export default Header;