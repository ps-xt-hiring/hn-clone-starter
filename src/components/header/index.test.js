import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Header from './index';

describe('Header component', () => {
  it('shallow renders without crashing', () => {
    shallow(<Header />);
  });

  it('renders top link', () => {
    const wrapper = shallow(<Header />);
    const text = wrapper.find('span.top').text();
    expect(text).toEqual('top');
  });

  it('renders new link', () => {
    const wrapper = shallow(<Header />);
    const text = wrapper.find('span.new').text();
    expect(text).toEqual('new');
  });

  it('header should not change b/w renders', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
