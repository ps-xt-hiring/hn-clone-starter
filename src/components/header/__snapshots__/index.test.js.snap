// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component header should not change b/w renders 1`] = `
<header
  className="header-container"
>
  <span
    className="header-items logo"
  >
    Y
  </span>
  <span
    className="header-items top active"
  >
    top
  </span>
  <span
    className="header-items new"
  >
    new
  </span>
</header>
`;
